src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -133,7 +137,9 @@ public bool PrepareForReuse(bool async)
             // If the read-ahead task is completed, then we've received either EOF or erroneous data the connection, so it's not usable.
             if (ReadAheadTaskHasStarted)
```

Comment:
could still this be race condition? e.g. It did not start it but it will before we get to the next statement.

Reply:
By the time you get to `PrepareForReuse`, the caller is the only thing with a reference to this connection.

It's possible that the read was already started and is transitioning to `Completed` right under you, but it can't transition from `NotStarted` => something else, or from something to => `NotStarted`.